### PR TITLE
TST: Removed import of itertools

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/groupby/test_function.py [hunk 1]
```diff
@@ -1,7 +1,6 @@
 import builtins
-import datetime as dt
+import datetime
```
Comment (simonjayhawkins) [inline pandas/tests/groupby/test_function.py]: can you separate this change into another PR for discussion. (keep this PR as removing itertools)

The reason is that we only have 7 occurrences of `import datetime as dt` in the codebase and there is some discrepancy between which datetime of datetime.datetime is aliased as dt

The more common pattern is like `from datetime import date, datetime, time, timedelta, tzinfo`

probably best with these cleanups, is to pick one type of change at a time.

**Review by simonjayhawkins — CHANGES_REQUESTED** [review]
Thanks @MomIsBestFriend generally lgtm ex datetime import.

File: pandas/tests/groupby/test_function.py (hunk 1 above)
Comment (simonjayhawkins) [reply pandas/tests/groupby/test_function.py]: > probably best with these cleanups, is to pick one type of change at a time.

and then after appropriate discussion can apply the changes across the complete codebase.

File: pandas/tests/groupby/test_function.py [hunk 2]
```diff
@@ -1296,36 +1307,32 @@ def __eq__(self, other):
...
-    result = grouped.size()
-    for key, group in grouped:
-        assert result[key] == len(group)
 
-    grouped = df.groupby("B")
-    result = grouped.size()
-    for key, group in grouped:
-        assert result[key] == len(group)
+@pytest.mark.parametrize("key", ["A", "B", ["A", "B"]])
+@pytest.mark.parametrize("sort", [True, False])
```
Comment (simonjayhawkins) [inline pandas/tests/groupby/test_function.py]: This brings the total number of sort parameterizations for [True, False] to 12. so maybe now worth considering creating a fixture for this. (as a follow-up)

**Review by simonjayhawkins — COMMENTED** [review]
a few minor nits, not blockers.

**Review by simonjayhawkins — APPROVED** [review]
Thanks @MomIsBestFriend lgtm pending green

File: pandas/tests/groupby/test_function.py (hunk 2 above)
Comment (MomIsBestFriend) [reply pandas/tests/groupby/test_function.py]: I think that is already a fixture for ```sort``` but it's only for ```[None, False]``` IIRC.

Comment (simonjayhawkins) [reply pandas/tests/groupby/test_function.py]: we can use different fixtures with the same name. The fixture used depends on the directory structure.

we have the relevant fixture in pandas\tests\reshape\test_concat.py, so just needs to be moved.

The sort fixture in pandas\tests\indexes\conftest.py has a caution note. maybe worth adding something like that.

BUT, if you do this, do in a follow-up, not here.

**Comment by simonjayhawkins** [comment]
Thanks @MomIsBestFriend

**Pull request merged**